src/libraries/System.Private.CoreLib/src/System/Numerics/Vector3.cs

```diff
@@ -465,19 +465,11 @@ public static Vector3 Create(ReadOnlySpan<float> values)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static Vector3 Cross(Vector3 vector1, Vector3 vector2)
         {
-            // This implementation is based on the DirectX Math Library XMVector3Cross method
-            // https://github.com/microsoft/DirectXMath/blob/master/Inc/DirectXMathVector.inl
-
-            Vector128<float> v1 = vector1.AsVector128Unsafe();
-            Vector128<float> v2 = vector2.AsVector128Unsafe();
-
-            Vector128<float> temp = Vector128.Shuffle(v1, Vector128.Create(1, 2, 0, 0)) * Vector128.Shuffle(v2, Vector128.Create(2, 0, 1, 0));
-
-            return Vector128.MultiplyAddEstimate(
-                -Vector128.Shuffle(v1, Vector128.Create(2, 0, 1, 0)),
-                 Vector128.Shuffle(v2, Vector128.Create(1, 2, 0, 0)),
-                 temp
-            ).AsVector3();
+            return Create(
+                (vector1.Y * vector2.Z) - (vector1.Z * vector2.Y),
+                (vector1.Z * vector2.X) - (vector1.X * vector2.Z),
+                (vector1.X * vector2.Y) - (vector1.Y * vector2.X)
+            );
```

Comment:
This will unacceptably regress perf. The "correct" fix is to use standard `multiply-add` (`a * b + c`) rather than `multiply-add estimate` (`mae(a, b, c)`)

Reply:
I did not found MultiplyAdd on Vector128 class, so I tried to convert it to two separate multiply/add. but I did found a code that does the following 


https://github.com/dotnet/runtime/blob/d513c4f964b68861885c94ac32debdaff9f6cef0/src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector128.cs#L634-L642

Should I do the same here ?